### PR TITLE
Modification of baskets and datasets

### PR DESCRIPTION
**Pull request by gacarrillor**
+ [x] Adapt Basket Manager and Create Basket dialogs, as well as their underlying models.
+ [x] Edit basket attributes.
+ [x] Change dataset of a basket.
+ [x] Delete basket.
+ [x] Delete dataset.

**Screenshots**

Edit basket:
image

-------

Depends on https://github.com/opengisch/QgisModelBakerLibrary/pull/107

### CODE REVIEW

**Comment by signedav** [comment]
Tested it on PG and it runs pretty smooth. Two minor GUI inputs:
- Maybe "edit" and "remove" icon can be depending on selection.
- The BID (OID Type) looks like expecting something. Can't we grey it out?
   image

Code review will follow...

File: QgisModelBaker/utils/ili2db_utils.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
+"""
+/***************************************************************************
+                              -------------------
+        begin                : 27.09.2024
```
Comment (signedav) [inline QgisModelBaker/utils/ili2db_utils.py:4]: My birthday gift :smiling_face_with_tear:

File: QgisModelBaker/utils/ili2db_utils.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+from QgisModelBaker.libs.modelbaker.iliwrapper import ilideleter
+from QgisModelBaker.libs.modelbaker.iliwrapper.ili2dbconfig import (
+    DeleteConfiguration,
+    Ili2DbCommandConfiguration,
+)
+from QgisModelBaker.libs.modelbaker.iliwrapper.ili2dbutils import JavaNotFoundError
+from QgisModelBaker.libs.modelbaker.utils.qt_utils import OverrideCursor
+
+
+class Ili2DbUtils(QObject):
```
Comment (signedav) [inline QgisModelBaker/utils/ili2db_utils.py:31]: Why's that not in the Library?

File: QgisModelBaker/gui/basket_manager.py [hunk 3]
```diff
@@ -17,51 +17,196 @@
...
+            layer.dataProvider().reloadData()
+            layer_tree_view.refreshLayerSymbology(layer.id())
+
+    def _do_delete_basket(self, basket_config):
+        # Keep original values just in case we need to go back to them
+        original_dataset_id = basket_config["dataset_t_id"]
+        original_dataset_name = basket_config["datasetname"]
+
+        # Create temporary dataset
+        tmp_dataset_id, tmp_dataset_name = "", "_tmp_dataset_tmp_"
```
Comment (signedav) [inline QgisModelBaker/gui/basket_manager.py:168]: Good idea for a workaround. Still I'll ask Claude...

Comment (gacarrillor) [reply QgisModelBaker/gui/basket_manager.py:168]: Yeap, implemented that way to have something functional within budget. Let's see what Claude tells us.
If it's supported by ili2db in the future, it'll be straightforward to switch to it.

File: QgisModelBaker/utils/ili2db_utils.py (hunk 2 above)
Comment (gacarrillor) [reply QgisModelBaker/utils/ili2db_utils.py:31]: I'll move it :)

Comment (gacarrillor) [reply QgisModelBaker/utils/ili2db_utils.py]: Done in https://github.com/opengisch/QgisModelBaker/pull/955/commits/805ee0e52f71ebdef39596a17800a77d9a5f641e and https://github.com/opengisch/QgisModelBakerLibrary/pull/107/commits/db232bd899618b02e79fb15595c3300aa2c6b90f

**Comment by gacarrillor** [comment]
> Tested it on PG and it runs pretty smooth. Two minor GUI inputs:
> 
>     * Maybe "edit" and "remove" icon can be depending on selection.
>     * The BID (OID Type) looks like expecting something. Can't we grey it out?

Addressed in the latest commits.

(After a while playing with colors...) Let's disable those read-only fields :) 
It's better for you as well?

image

**Comment by signedav** [comment]
Good to merge after a new library release...

**Pull request merged**